Features/bootstrap/MainContext.php

```diff
@@ -0,0 +1,55 @@
+<?php
+
+use QATools\BehatExtension\Context\QAToolsContext;
+
+class MainContext extends QAToolsContext
+{
+
+	/**
+	 * @Given I'm on ':page'
+	 */
+	public function openPage($page)
+	{
+		$this->page = $this->pageFactory->getPage($page);
+		$this->page->open();
+	}
+
+	/**
+	 * @When I login with ':username' and ':password'
+	 */
+	public function login($username, $password)
+	{
+		$this->page->login($username, $password);
+	}
+
+	/**
+	 * @Then I should be on ':page'
+	 */
+	public function pageOpen($page)
+	{
+		$this->page = $this->pageFactory->getPage($page);
```

Comment:
I haven't seen any best practices yet where to assign what. I just did it here cause I didn't know any good way to detect the successful or failing login in `login` that why I used `@Then` to fetch and store the correct page.